src/dhcpmon/src/dhcp_device.c

```diff
@@ -152,6 +196,45 @@ static void handle_dhcp_option_53(dhcp_device_context_t *context,
     }
 }
 
+/**
+ * @code handle_dhcpv6_option(context, dhcp_option, dir);
+ *
+ * @brief handle the logic related to DHCPv6 option
+ *
+ * @param context       Device (interface) context
+ * @param dhcp_option   pointer to DHCP option buffer space
+ * @param dir           packet direction
+ *
+ * @return none
+ */
+static void handle_dhcpv6_option(dhcp_device_context_t *context,
+                                  const u_char dhcp_option,
+                                  dhcp_packet_direction_t dir)
+{
+    switch (dhcp_option)
+    {
+    case DHCPv6_MESSAGE_TYPE_SOLICIT:
+    case DHCPv6_MESSAGE_TYPE_REQUEST:
+    case DHCPv6_MESSAGE_TYPE_CONFIRM:
+    case DHCPv6_MESSAGE_TYPE_RENEW:
+    case DHCPv6_MESSAGE_TYPE_REBIND:
+    case DHCPv6_MESSAGE_TYPE_RELEASE:
+    case DHCPv6_MESSAGE_TYPE_DECLINE:
+    case DHCPv6_MESSAGE_TYPE_ADVERTISE:
+    case DHCPv6_MESSAGE_TYPE_REPLY:
+    case DHCPv6_MESSAGE_TYPE_RECONFIGURE:
+    case DHCPv6_MESSAGE_TYPE_INFORMATION_REQUEST:
+    case DHCPv6_MESSAGE_TYPE_RELAY_FORWARD:
```

Comment:
can we get rid of relay-forward and relay-reply message types and count inner message against DHCPv6 message type?

Reply:
@tahmed-dev I changed the code to fit your suggestion, please review.
Thanks

Reply:
Thanks. I noticed now that the parser will point to inner type of encapped packet. Can we remove the Relay Forward and Relay Reply from this case and let them fall through to the default section as this should not be expected.

Reply:
Done.